erdjs/src/dapp/walletConnectProvider.ts

```diff
@@ -0,0 +1,198 @@
+            accounts: [account],
+        } = params[0];
+
+        this.loginAccount(account);
+    };
+
+    onDisconnect = async (error: any) => {
+        if (error) {
+            throw error;
+        }
+
+        this.dispatchEvent(this.onWalletConnectDisconect);
+    };
+
+    /**
+     * Mocks a logout request by returning true
+     */
+    async logout(): Promise<boolean> {
+        if (!this.walletConnector) {
+            throw new Error("Wallet Connect not initialised, call init() first");
+        }
+        this.walletConnector.killSession();
+        return true;
+    }
+
+    /**
+     * Fetches current selected ledger address
+     */
+    async getAddress(): Promise<string> {
+        return this.address;
+    }
+
+    /**
+     * Signs and sends a transaction. Returns the transaction hash
+     * @param transaction
+     */
+    async sendTransaction(transaction: Transaction): Promise<Transaction> {
+        if (!this.walletConnector) {
+            throw new Error("Wallet Connect not initialised, call init() first");
+        }
+
+        const address = await this.getAddress();
+        const sig = await this.signTransaction("erd_sign", this.prepareWalletConnectMessage(transaction, address));
+        if (sig && sig.signature) transaction.applySignature(new Signature(sig.signature), new Address(address));
+
+        await transaction.send(this.provider);
+
+        return transaction;
+    }
+
+    private async signTransaction(method: any, params: any): Promise<any> {
+        if (!this.walletConnector) {
+            throw new Error("Wallet Connect not initialised, call init() first");
+        }
+        if (this.walletConnector.connected) {
+            const response = await this.walletConnector.sendCustomRequest({
+                method,
+                params,
+            });
+
+            return response;
+        } else {
+            return false;
+        }
+    }
+
+    private prepareWalletConnectMessage(transaction: Transaction, address: string): any {
+        return {
+            nonce: transaction.getNonce().valueOf(),
+            from: address,
+            to: transaction.getReceiver().toString(),
+            amount: transaction.getValue().toString(),
+            gasPrice: transaction
+                .getGasPrice()
+                .valueOf()
+                .toString(),
+            gasLimit: transaction
+                .getGasLimit()
+                .valueOf()
+                .toString(),
+            data: Buffer.from(
+                transaction
+                    .getData()
+                    .toString()
+                    .trim()
+            ).toString(),
+            chainId: transaction.getChainID().valueOf(),
+            version: transaction.getVersion().valueOf(),
+        };
+    }
+
+    private canTransformToPublicKey(address: string): boolean {
+        try {
+            const checkAddress = new Address(address);
+            return Boolean(checkAddress.bech32());
```

Comment:
Don't we want `return true` here? (not sure)

Reply:
updated